python/pyarrow/tests/test_table.py

```diff
@@ -239,23 +239,65 @@ def test_concat_tables():
 
 
 def test_table_pandas():
-    data = [
```

Comment:
Can you move these tests into `python/pyarrow/tests/test_convert_pandas.py`? 

As they depend on pandas, they should be there so that these tests here can run without pandas being installed.